zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/RecallMailServiceImpl.java

```diff
@@ -0,0 +1,209 @@
+package fr.openent.zimbra.tasks.service.impl;
+
+import fr.openent.zimbra.core.enums.ActionType;
+import fr.openent.zimbra.core.enums.ErrorEnum;
+import fr.openent.zimbra.core.enums.TaskStatus;
+import fr.openent.zimbra.helper.ConfigManager;
+import fr.openent.zimbra.helper.MessageHelper;
+import fr.openent.zimbra.model.message.Message;
+import fr.openent.zimbra.model.message.RecallMail;
+import fr.openent.zimbra.model.soap.SoapMessageHelper;
+import fr.openent.zimbra.model.task.RecallTask;
+import fr.openent.zimbra.service.impl.NotificationService;
+import fr.openent.zimbra.service.impl.RecipientService;
+import fr.openent.zimbra.tasks.service.DbRecallMail;
+import fr.openent.zimbra.tasks.service.RecallMailService;
+import fr.openent.zimbra.service.StructureService;
+import fr.openent.zimbra.service.data.Neo4jZimbraService;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.commons.lang3.NotImplementedException;
+import org.entcore.common.user.UserInfos;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static fr.openent.zimbra.model.constant.ZimbraConstants.ADDR_TYPE_FROM;
+
+public class RecallMailServiceImpl implements RecallMailService {
+    public static ConfigManager appConfig;
+    private final RecallQueueServiceImpl recallQueueService;
+    private final NotificationService notificationService;
+    private final StructureService structureService;
+    private final DbRecallMail dbMailService;
+    private final RecipientService recipientService;
+    private final Neo4jZimbraService neo4jService;
+
+    private final static int BATCH_SIZE = 100;
```

Comment:
static avant final dans la convention de java

```suggestion
    private static final int BATCH_SIZE = 100;
```